File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsIntegrationsAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

// Status-only models for M14.8 — secrets are intentionally not modeled.

@Serializable
data class LtiRegistrationsResponse(
    val parentPlatforms: List<LtiParentPlatform> = emptyList(),
    val externalTools: List<LtiExternalTool> = emptyList(),
)

@Serializable
data class LtiParentPlatform(
    val id: String,
    val name: String = "",
    val clientId: String = "",
    val platformIss: String = "",
    val active: Boolean = false,
)

@Serializable
data class LtiExternalTool(
    val id: String,
    val name: String = "",
    val clientId: String = "",
    val toolIssuer: String = "",
    val active: Boolean = false,
)

@Serializable
data class LtiActiveBody(val active: Boolean)

@Serializable
data class ScimTokensResponse(val tokens: List<ScimTokenRow>? = null)

@Serializable
data class ScimTokenRow(
    val id: String,
    val institutionId: String = "",
    val label: String = "",
    val createdAt: String = "",
    val revokedAt: String? = null,
)

@Serializable
data class ScimEventsResponse(val events: List<ScimEventRow>? = null)

@Serializable
data class ScimEventRow(
    val id: String,
    val operation: String = "",
    val scimResource: String = "",
    val userEmail: String? = null,
    val createdAt: String = "",
)

@Serializable
data class PlatformScimFlag(val scimEnabled: Boolean? = null)

/** Secret-free cloud provider status for mobile admin. */
@Serializable
data class CloudProviderStatus(
    val provider: String,
    val enabled: Boolean = false,
    val updatedAt: String? = null,
)

@Serializable
data class CloudProviderEnabledBody(val enabled: Boolean)

@Serializable
data class LrsEndpointStatus(
    val id: String,
    val label: String = "",
    val endpointUrl: String = "",
    val authType: String = "",
    val username: String? = null,
    val enabled: Boolean = false,
    val hasPassword: Boolean? = null,
    val hasOauthSecret: Boolean? = null,
    val updatedAt: String? = null,
)

@Serializable
data class LrsEnabledBody(val enabled: Boolean)

@Serializable
data class OerProviderStatus(
    val provider: String,
    val enabled: Boolean = false,
    val updatedAt: String? = null,
)

@Serializable
data class OerProviderEnabledBody(val enabled: Boolean)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsIntegrationsAdmin.swift
import Foundation

// MARK: - Integrations & provisioning admin models (M14.8)

// Status-only models: secrets (cloud keys, SCIM bearer tokens, LRS passwords) are never decoded.

struct LtiRegistrationsResponse: Decodable {
    var parentPlatforms: [LtiParentPlatform]
    var externalTools: [LtiExternalTool]
}

struct LtiParentPlatform: Decodable, Identifiable, Hashable {
    var id: String
    var name: String
    var clientId: String
    var platformIss: String
    var active: Bool
}

struct LtiExternalTool: Decodable, Identifiable, Hashable {
    var id: String
    var name: String
    var clientId: String
    var toolIssuer: String
    var active: Bool
}

struct LtiActiveBody: Encodable {
    var active: Bool
}

struct ScimTokensResponse: Decodable {
    var tokens: [ScimTokenRow]?
}

struct ScimTokenRow: Decodable, Identifiable, Hashable {
    var id: String
    var institutionId: String
    var label: String
    var createdAt: String
    var revokedAt: String?
}

struct ScimEventsResponse: Decodable {
    var events: [ScimEventRow]?
}

struct ScimEventRow: Decodable, Identifiable, Hashable {
    var id: String
    var operation: String
    var scimResource: String
    var userEmail: String?
    var createdAt: String
}

/// Platform settings subset used only for SCIM feature gating.
struct PlatformScimFlag: Decodable {
    var scimEnabled: Bool?
}

/// Secret-free cloud provider status for mobile admin.
struct CloudProviderStatus: Decodable, Identifiable, Hashable {
    var provider: String
    var enabled: Bool
    var updatedAt: String?

    var id: String { provider }
}

struct CloudProviderEnabledBody: Encodable {
    var enabled: Bool
}

struct LrsEndpointStatus: Decodable, Identifiable, Hashable {
    var id: String
    var label: String
    var endpointUrl: String
    var authType: String
    var username: String?
    var enabled: Bool
    var hasPassword: Bool?
    var hasOauthSecret: Bool?
    var updatedAt: String?
}

struct LrsEnabledBody: Encodable {
    var enabled: Bool
}

struct OerProviderStatus: Decodable, Identifiable, Hashable {
    var provider: String
    var enabled: Bool
    var updatedAt: String?

    var id: String { provider }
}

struct OerProviderEnabledBody: Encodable {
    var enabled: Bool
}
